content/kb/troubleshooting-volume-filesystem-corruption.md

```diff
@@ -48,3 +51,36 @@ Longhorn cannot fix this automatically. You will need to resolve this manually w
 7. Detach the volume from the UI.
 8. Scale up the workload.
 
+### For some older Linux distribution versions and Longhorn volumes with ext4 filesystems
+
+In the CSI flow, the Longhorn CSI plugin creates a file system on a new volume using a `mkfs.ext4` built into its
+container. The `fsck.ext4` available in some older Linux distributions may not support all features this filesystem is
+created with, resulting in the following error:
+
+```
+-> fsck.ext4 /dev/longhorn/pvc-c7152ef5-55c7-43ce-a35e-dac69d2be591 
+e2fsck 1.42.9 (28-Dec-2013)
+/dev/longhorn/pvc-c7152ef5-55c7-43ce-a35e-dac69d2be591 has unsupported feature(s): metadata_csum
+e2fsck: Get a newer version of e2fsck!
+```
+
+If possible, upgrade your `e2fsprogs` to a later version. If not possible (e.g. on CentOS 7 or RHEL 7), the
+`instance-manager` or `instance-manager-e` container has an updated `fsck.ext4` built in and has access to attached
+Longhorn volumes.
+
+Follow steps 1-3 from above.
+
+4. Exec into the `instance-manager` or `instance-manager-e` pod running on the node the volume is attached to.  
+   `kubectl exec -it -n longhorn-system instance-manager-<additional-characters> -- bash`
+
+Follow steps 5-8 from above.
```

Comment:
```suggestion
5. Find the block device corresponding to the Longhorn volume under `/dev/longhorn/<volume-name>`.
6. Use a file system check tool to repair the file system.
Examples:
- [`fsck`](https://man7.org/linux/man-pages/man8/fsck.8.html): Checks and optionally repairs Linux file systems (including ext4)
- [`xfs_repair`](https://man7.org/linux/man-pages/man8/xfs_repair.8.html): Repairs XFS file systems
7. On the Longhorn UI, detach the volume.
8. Scale up the workload.
```

Reply:
To provide a better user experience, we should list all relevant steps and related information within the section. Self-contained procedures are easier to follow.